src/vanilla/Model/CodeModelTS.cs

```diff
@@ -391,6 +394,110 @@ public string RequiredConstructorParametersTS
             }
         }
 
+        public virtual string GenerateConstructor(string superParameterList, Action<TSBlock> guardChecks = null, Action<TSBlock> implementation = null)
+        {
+            TSBuilder builder = new TSBuilder();
+            var clientOptionType = OptionalParameterTypeForClientConstructor == GetServiceClientOptionsName()
+                ? "msRest." + GetServiceClientOptionsName()
+                : OptionalParameterTypeForClientConstructor;
+
+            string parameterList = (!string.IsNullOrEmpty(RequiredConstructorParametersTS) ? RequiredConstructorParametersTS + ", " : "") + "options?: " + clientOptionType;
+
+            builder.Constructor(parameterList, superParameterList, guardChecks, implementation);
+
+            return builder.ToString();
+        }
+
+        public virtual string GenerateClientConstructor()
+        {
+            string superParameterList = (!string.IsNullOrEmpty(RequiredConstructorParameters) ? RequiredConstructorParameters + ", " : "") + "options";
+            return this.GenerateConstructor(superParameterList, null, GenerateOperationInstantiating);
+        }
+
+        public virtual string GenerateContextConstructor(string emptyLine)
+        {
+            Action<TSBlock> guardCheck = block =>
+            {
+                foreach (var param in RequiredParameters)
+                {
+                    block.If($"{param.Name} === null || {param.Name} === undefined", then =>
+                    {
+                        then.Line($"throw new Error(\"\'{param.Name}\' cannot be null.\");");
+                    });
+                }
+
+                if (RequiredParameters.Any())
+                {
+                    block.Line(emptyLine);
+                }
+
+                block.If("!options", then =>
+                {
+                    then.Line("options = {};");
```

Comment:
Replace this with an `AssignmentObject("options")` call. If AssignmentObject() doesn't already exist, then I recommend creating it.